Sources/StreamChatUI/CommonViews/ContainerStackView.swift

```diff
@@ -437,6 +431,27 @@ public class ContainerStackView: UIView {
         spacingConstraintsByView[subview]?.resetTemporaryConstant()
         layoutIfNeeded()
     }
+    
+    /// Update constraints related to `Distribution.equal`.
+    private func updateDistributionEqualConstraints() {
+        NSLayoutConstraint.deactivate(equalConstraints)
+        equalConstraints = []
+        // Make the arranged subviews all with the same size in case of equal distribution
+        guard distribution == .equal else { return }
+        let visibleSubviews = subviews.filter { !$0.isHidden }
+        zip(visibleSubviews, visibleSubviews.dropFirst()).forEach { lView, rView in
+            if axis == .horizontal {
+                equalConstraints.append(
+                    lView.widthAnchor.constraint(equalTo: rView.widthAnchor)
+                )
+            } else {
+                equalConstraints.append(
+                    lView.heightAnchor.constraint(equalTo: rView.heightAnchor)
+                )
+            }
+        }
+        NSLayoutConstraint.activate(equalConstraints)
+    }
```

Comment:
We already have a pattern here where we story the constraints in a dictionary so we can selectively disable/enable them if needed. Why not do it the same here? This feels a bit hacky + it does a lot of unnecessary work. We should be able to do just something similar to this:
```swift
distributionConstraintByView[subview]?.isActive = true / false

// or even...

distributionConstraintsByView[subview]?.forEach { $0.isActive = true / false }
```

Reply:
it is not that simple in this case. Let's say I have three views: A, B, C. Constraints for the same width are created between A-B and B-C. I hide subview B. Therefore, I should create a new constraint between A and C.

Reply:
I thought recomputing them altogether would be more straightforward - or do you think we should try to be "smarter" here? Even in the case I presented, if you have a lot of views, most of them will have the same constraints before and after